ntoskrnl/mm/ARM3/miarm.h

```diff
@@ -2428,21 +2428,29 @@ FORCEINLINE
 BOOLEAN
 MiSynchronizeSystemPde(PMMPDE PointerPde)
 {
-    MMPDE SystemPde;
     ULONG Index;
 
     /* Get the Index from the PDE */
     Index = ((ULONG_PTR)PointerPde & (SYSTEM_PD_SIZE - 1)) / sizeof(MMPTE);
+    if (PointerPde->u.Hard.Valid != 0)
+    {
+        NT_ASSERT(PointerPde->u.Long == MmSystemPagePtes[Index].u.Long);
+        return TRUE;
+    }
+
+    if (MmSystemPagePtes[Index].u.Hard.Valid == 0)
```

Comment:
When this MiSync... function is called, there is no way that the value stored in `MmSystemPagePtes[Index]` can change in between there and when it's used below in MI_WRITE_VALID_PDE() ?

Reply:
This caller holds the kernel address space lock. So we can hope that is a no.